source/beamline/BioXAS/BioXASCarbonFilterFarmActuatorControl.h

```diff
@@ -38,100 +35,73 @@ class BioXASCarbonFilterFarmActuatorControl : public AMCompositeControl
-	static Window::Selection stringToWindow(const QString &string);
-
-	/// Returns the window corresponding to the given enum index.
-	static Window::Selection window(double index);
+	static double stringToWindow(const QString &string);
 
 	/// Returns the window corresponding to the given position. Returns Window::None if no window found.
-	Window::Selection windowAtPosition(double position) const;
+	double windowAtPosition(double position) const;
 	/// Returns the position of the given window. Returns 0 if no position was found.
-	double positionOfWindow(Window::Selection window) const;
+	double positionOfWindow(double window) const;
 
 public slots:
-	/// Sets the new window setpoint and moves the actuator.
-	virtual FailureExplanation move(double setpoint);
-
 	/// Sets a window to position mapping.
-	void setWindowPosition(Window::Selection window, double position);
+	void setWindowPosition(double window, double position);
+	/// Sets the position control.
+	void setPositionControl(AMControl *newControl);
+	/// Sets the status control.
+	void setStatusControl(AMControl *newControl);
 
 protected slots:
-	/// Sets the active window.
-	void setWindow(double newWindow);
-	/// Sets the desired window setpoint.
-	void setWindowSetpoint(double newWindow);
-	/// Sets the move in progress.
-	void setMoveInProgress(bool isMoving);
-
-	/// Handles emitting the appropriate signals when the move action has started.
-	void onMoveStarted();
-	/// Handles emitting the appropriate signals and performing action cleanup when the move action is cancelled.
-	void onMoveCancelled(QObject *action);
-	/// Handles emitting the appropriate signals and performing action cleanup when the move action fails.
-	void onMoveFailed(QObject *action);
-	/// Handles emitting the appropriate signals and performing action cleanup when the move action succeeds.
-	void onMoveSucceeded(QObject *action);
-
-	/// Updates the active window, based in the current position and the available mappings.
-	void updateWindow();
+	/// Updates the connected state.
+	virtual void updateConnected();
+	/// Updates the current value.
+	virtual void updateValue();
+	/// Updates the 'is moving' state.
+	virtual void updateIsMoving();
 
 protected:
 	/// Returns a new action that moves the actuator to the desired window setpoint. Returns 0 if unable to move or if the setpoint is invalid.
-	AMAction3* createMoveAction(double setpoint);
+	virtual AMAction3* createMoveAction(double windowSetpoint);
 
 protected:
-	/// The active window.
-	double value_;
-	/// The active window setpoint.
-	double setpoint_;
-	/// The current move state, true if this control has initiated a move.
-	bool moveInProgress_;
-	/// The mapping between window and position.
-	QMap<Window::Selection, double> positionMap_;
+	/// The mapping between window enum and physical window position.
+	QMap<double, double> positionMap_;
+
 	/// The position control.
-	AMPVControl *position_;
+	AMControl *position_;
```

Comment:
I think it's best to keep them separate for now, as the status pv reports whether or not the position control is in a valid position (not it's moving status).